### PR TITLE
WIP - ARM

### PR DESCRIPTION
**Pull request by tgiphil**

### CODE REVIEW

File: Source/Docs/settings-options.rst [hunk 1]
```diff
@@ -201,7 +201,9 @@ Application Location Settings
    :widths: 50, 200
 
     AppLocation.Bochs,Location of the BOCHS application
-    AppLocation.Qemu,Location of the QEMU application
+    AppLocation.QemuX86,Location of the QEMU application for x86 (i386)
```
Comment (charsleysa) [inline Source/Docs/settings-options.rst:204]: what about x64?

File: Source/Mosa.Utility.Disassembler/Disassembler.cs [hunk 2]
```diff
@@ -28,6 +28,7 @@ public Disassembler(string platform)
 		arch = platform.ToLowerInvariant() switch
 		{
 			"armv8a32" => new Arm32Architecture(services, "arm32", options),
+			"arm64" => new Arm32Architecture(services, "arm64", options),
```
Comment (charsleysa) [inline Source/Mosa.Utility.Disassembler/Disassembler.cs:31]: shouldn't this be Arm64Architecture?

File: Source/Mosa.Utility.Launcher/Starter.cs [hunk 3]
```diff
@@ -257,18 +259,40 @@ public Process LaunchVM()
...
+
+			case "armv8a32":
+				qemuApp = MosaSettings.QemuARM32App;
+				uefi = $" -drive if=pflash,format=raw,readonly=on,file={Quote(MosaSettings.QemuEdk2ARM)}";
+				arg.Append(" -cpu arm1176");
+				break;
+
+			case "arm64":
+				qemuApp = MosaSettings.QemuARM64App;
+				arg.Append(" -cpu cortex-a7");
```
Comment (charsleysa) [inline Source/Mosa.Utility.Launcher/Starter.cs:291]: what about uefi?

File: Source/Mosa.Utility.Launcher/Starter.cs [hunk 4]
```diff
@@ -257,18 +259,40 @@ public Process LaunchVM()
...
+			case "x86":
+				qemuApp = MosaSettings.QemuX86App;
+				uefi = $" -drive if=pflash,format=raw,readonly=on,file={Quote(MosaSettings.QemuEdk2X86)}";
+				arg.Append(" -cpu qemu32,+sse4.1,abm,bmi1,bmi2,popcnt");
+				break;
+
+			case "x64":
+				qemuApp = MosaSettings.QemuX64App;
+				uefi = $" -drive if=pflash,format=raw,readonly=on,file={Quote(MosaSettings.QemuEdk2X64)}";
+				break;
```
Comment (charsleysa) [inline Source/Mosa.Utility.Launcher/Starter.cs:281]: what about cpu features?

**Review by charsleysa — CHANGES_REQUESTED** [review]

File: Source/Docs/settings-options.rst (hunk 1 above)
Comment (tgiphil) [reply Source/Docs/settings-options.rst:204]: Added.

File: Source/Mosa.Utility.Disassembler/Disassembler.cs (hunk 2 above)
Comment (tgiphil) [reply Source/Mosa.Utility.Disassembler/Disassembler.cs]: Fixed.

File: Source/Mosa.Utility.Launcher/Starter.cs (hunk 3 above)
Comment (tgiphil) [reply Source/Mosa.Utility.Launcher/Starter.cs:291]: Fixed.

File: Source/Mosa.Utility.Launcher/Starter.cs (hunk 4 above)
Comment (tgiphil) [reply Source/Mosa.Utility.Launcher/Starter.cs:280]: Fixed.

**Review by charsleysa — APPROVED** [review]
LGTM

**Pull request merged**